pkg/util/util.go

```diff
@@ -404,7 +404,7 @@ type TimeoutFunc func() (err error)
 // WaitUntilTimeout waits for the exec function to complete or return timeout error
 func WaitUntilTimeout(timeout time.Duration, execFunc ExecFunc, timeoutFunc TimeoutFunc) error {
 	// Create a channel to receive the result of the azcopy exec function
-	done := make(chan bool)
+	done := make(chan bool, 1)
```

Comment:
why set buffer size as 1 would fix the issue?

Reply:
The write operation to a channel in Go is blocking unless there is a receiver waiting for it, this means after `WaitUntilTimeout` exit, the goroutine will remain stuck at `done <- true` forever.
After changed to buffered channel, even if there is no receiver, done <- true can successfully write and continue.